### DISCUSSION TITLE
Is Apple Silicon is supported by Harman AIR

### DISCUSSION THREAD

**Discussion by zabdoul60460** [airsdk/Adobe-Runtime-Support#2107] [Q&A]
Hi,
I used VS Code to compile air application for Mac OS 64 bits CPU (intel, M1, M2).
Is Apple Silicon is supported by Harman AIR.
If yes how to configure config file ? If not how to do when rosetta 2 will be stopped.
Best regards.

**Comment by ajwfrost** [comment]
Hi

Yes, Apple silicon is supported automatically, for a while now. The AIR runtimes on macOS use universal binaries and when you generate a new application for mac, it will be the same.

thanks

**Comment by zabdoul60460** [comment]
Thanks,
In this case could explain why I have to to check 'Start with rosetta', if not, it's impossible to start air application !
To generate my application, I work with VSCode and AS3Mxml. Should I specify a particular option on asconfig.json ?
Thanks for helping.
Best regards

**Reply by ajwfrost** [reply]
So the one part of AIR that doesn't have arm64 support is the WebKit.dylib library - which we discontinued after 33.1.1.743 for this very reason. If you're using an HTMLControl or a non-native StageWebView, then this would try to load in the WebKit.dylib library and if your application was started as an arm64 process, it would then fail.

Options here would be
1) switch to using the native StageWebView class (may need to refactor your app...) and then you can continue using later AIR SDK versions
2) continue with 33.1.1.743 and use something like "lipo" to strip out the arm64 support in your binary, thus making it always launch in x86_64 mode with Rosetta without any user action needed. This would be a step needed to be done manually after generating the app, there's no support for it in the build tools.

thanks

**Reply by zabdoul60460** [reply]
In my application, I never used HTMLControl. What is for you a non-native StageWebView ?
Best regards

**Reply by ajwfrost** [reply]
Hmm.. curious. For non-native `StageWebView` it means the default constructor: you need to pass `true` in as the first parameter to make it a native one (i.e. using the operating system capability).
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/StageWebView.html#StageWebView()

Can I check what version of the AIR SDK you're using? It's always possible perhaps that this is a different issue .. sometimes we've found odd issues around security that seem to apply differently between the variants. If you want to send us the .app bundle we can try seeing what it may be..?

thanks

**Comment by zabdoul60460** [comment]
Curious indeed ...
I have any StageWebView or HTMLControl on my code.
I work with Apache Flex 4.16.1 and AIR 33.1. But I don't know how to find the subversion.
Do you think that if I'm using "bundle" instead of "native" when targeting desktop it could solve my problem ?
Best regards.

**Reply by ajwfrost** [reply]
Ah, so if you're creating a "native" packager then this doesn't actually include the runtime, instead it will need to be installed onto a machine that already has the AIR runtime installed as a shared library... I thought that this would still result in the universal binary though - but it may depend which version you have there!

Best way to check the exact version of any AIR runtime is to open Adobe Scout, and have this running in the background. Once the app has started, you can look at the Session Info and look for the runtime version..

thanks

**Reply by zabdoul60460** [reply]
I updated the AIR runtime from 33.1.1.743 to  33.1.1.932 but the problem is the same. Do you know why ?
For memory, I run my application on a device using Apple Silicon (M1 / ARM hardware). 
Best regards

**Reply by ajwfrost** [reply]
I would expect that there's some dependency upon the HTMLLoader class, which is then going back to the fact that the webkit library isn't available on M1 hardware. If you're using Flex then it's possible there's some component from that which is using it in the background..

So does it still work if you select the Rosetta option?

**Reply by zabdoul60460** [reply]
yes it work with rosetta option. How found dependency of HTMLLoader ? Indeed my project come from Flex to VSCode.
Best regards

**Reply by zabdoul60460** [reply]
the only html keyword is for richText property !

**Reply by zabdoul60460** [reply]
Other strange aspect, on application report my application is Universal type but stat only if Roseta option is selected !

**Reply by ajwfrost** [reply]
Some other Flex component may be using an HTMLLoader in the background perhaps? I'm wondering whether we would be able to programmatically detect/alert people about this but it's a bit of a challenge..

Yes the binaries would be classed as "universal" because the main executable supports both x86_64 and arm64; it launches by default in arm64 and then fails to load the dylib which only has x86_64. Setting it to run in Rosetta means it will launch already in x86_64 mode and the dylib can then be successfully loaded/run. If there was only an x86_64 binary within the main executable, then you wouldn't need that step (which you can do via `lipo`..)

thanks
